src/OpenTelemetry/Metrics/MetricReader.cs

```diff
@@ -226,6 +226,19 @@ public bool Shutdown(int timeoutMilliseconds = Timeout.Infinite)
         /// <inheritdoc/>
         public void Dispose()
         {
+            if (this.metrics != null)
+            {
+                foreach (var metric in this.metrics)
+                {
+                    if (metric == null)
+                    {
+                        break;
+                    }
+
+                    metric.AggStore.RemoveMeasurementDroppedCallbacks();
```

Comment:
This would not work for multiple `MetricReader` scenarios.

Reply:
This should work on a combination of `Instrument` and `MetricReader`. So we should also remove the entry from the dictionary when the instrument's `Meter` is disposed.

Reply:
Added the removal on MeasureComplete callbacks.